generated/nirfmxwlan/nirfmxwlan_service.cpp

```diff
@@ -2964,6 +3055,49 @@ namespace nirfmxwlan_grpc {
     }
   }
 
+  //---------------------------------------------------------------------
+  //---------------------------------------------------------------------
+  ::grpc::Status NiRFmxWLANService::OFDMModAccCfgNReferenceWaveforms(::grpc::ServerContext* context, const OFDMModAccCfgNReferenceWaveformsRequest* request, OFDMModAccCfgNReferenceWaveformsResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      auto instrument_grpc_session = request->instrument();
+      niRFmxInstrHandle instrument = session_repository_->access_session(instrument_grpc_session.id(), instrument_grpc_session.name());
+      char* selector_string = (char*)request->selector_string().c_str();
+      auto x0 = const_cast<float64*>(request->x0().data());
+      auto dx = const_cast<float64*>(request->dx().data());
+      auto reference_waveform = convert_from_grpc<NIComplexSingle>(request->reference_waveform());
+      auto total_length = std::accumulate(request->reference_waveform_lengths().cbegin(), request->reference_waveform_lengths().cend(), 0);
+
+      if (total_length != request->reference_waveform_size()) {
+        return ::grpc::Status(::grpc::INVALID_ARGUMENT, "The total size of reference_waveform did not match the exected size from reference_waveform_lengths");
+      }
+
+      auto reference_waveform_lengths = const_cast<int32*>(reinterpret_cast<const int32*>(request->reference_waveform_lengths().data()));
+      auto array_size_determine_from_sizes = std::array<int, 3>
+      {
+        request->x0_size(),
+        request->dx_size(),
+        request->reference_waveform_lengths_size()
```

Comment:
@astarche Dane mentioned to me he went with lengths because this generated code got choked up it whenever it was named _size().

Reply:
We've updated to `sizes`